db/migrate_cards/20160413193247_source_type_set_cards.rb

```diff
@@ -0,0 +1,18 @@
+# -*- encoding : utf-8 -*-
+
+class SourceTypeSetCards < Card::Migration
+  def up
+    %w(File Link Text).each do |name|
+      create_source_type_set name
+    end
+  end
+
+  def create_source_type_set name
+    create_card!(
+      name: "#{name}+*source type",
+      type_id: Card::SetID,
+      content: '{"type":"source",' +
+               %("right_plus":["*source type", {"refer_to":"#{name}"}]})
+    )
+  end
+end
```

Comment:
this creates a rule where "File+*source type", "Link+*source type" and "Text+*source type" are sets so the answer is no, it doesn't override and break things. The other approaches I used before with "source type+*source type+*type plus right" and "*source type+*right" broke things.